lib/elixir/lib/map_set.ex

```diff
@@ -177,35 +154,12 @@ defmodule MapSet do
 
       iex> MapSet.symmetric_difference(MapSet.new([1, 2, 3]), MapSet.new([2, 3, 4]))
       MapSet.new([1, 4])
+
   """
   @doc since: "1.14.0"
   @spec symmetric_difference(t(val1), t(val2)) :: t(val1 | val2) when val1: value, val2: value
-  def symmetric_difference(%MapSet{map: map1}, %MapSet{map: map2}) do
-    {small, large} = order_by_size(map1, map2)
-
-    map =
-      large
-      |> :maps.iterator()
-      |> :maps.next()
-      |> disjointer(small, [])
-
-    %MapSet{map: map}
-  end
-
-  defp disjointer(:none, small, list) do
-    list |> Map.from_keys(@dummy_value) |> Map.merge(small)
-  end
-
-  defp disjointer({key, _val, iter}, small, list) do
-    if is_map_key(small, key) do
-      iter
-      |> :maps.next()
-      |> disjointer(Map.delete(small, key), list)
-    else
-      iter
-      |> :maps.next()
-      |> disjointer(small, [key | list])
-    end
+  def symmetric_difference(%MapSet{set: set1} = map_set1, %MapSet{set: set2} = _map_set2) do
+    %{map_set1 | set: :sets.union(:sets.subtract(set1, set2), :sets.subtract(set2, set1))}
```

Comment:
This is the only part I’m thinking about: is this ok? Or should we use `sets:fold/3` to do this more similarly to what we were doing before?

Reply:
I think we should use sets fold, yes. It should be more efficient.

Reply:
@josevalim cool, updated!